### PR TITLE
Patched os.path.exists in testBuildTemplateApps

### PR DESCRIPTION
**Pull request by orangejenny**
## Technical Summary
This test is slow enough to intermittently fail.

Its original intent was to protect against issues when the template app json files got updated.

Patching this cuts off multimedia handling, which is the slow part, because each piece of multimedia is fetched via an HTTP request: https://github.com/dimagi/commcare-hq/blob/57f1f8530359a76422b8d6871e51dd2f957c2991/corehq/apps/app_manager/views/apps.py#L522-L541

### Safety story
It's a test.

### Automated test coverage

It's a test.

### QA Plan

No

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

**Review by gherceg — COMMENTED** [review]
Potential ~~nit~~ issue: As a followup to Matt's comment it also looks like this test doesn't sufficiently do what it sets out to based on "Tests that these apps successfully build" since `load_app_from_slug` returns the app object if successful, and None if it encounters some exceptions (like the AppValidationError). Though this is also a result of how broadly scoped the test is. 

Nit: Maybe it's the case that there is a specific exception that is raised that prompted this test in the first place, but it would be nice have this be more explicit. Something like:
```
try:
  load_app_from_slug(self.domain, 'username', slug)
except SpecificException:
  self.fail(...)
  ```

File: corehq/apps/app_manager/tests/test_apps.py [hunk 1]
```diff
@@ -262,7 +262,8 @@ def testBuildImportedApp(self):
         self._check_has_build_files(copy, self.min_paths)
         self._check_legacy_odk_files(copy)
 
-    def testBuildTemplateApps(self):
+    @patch('os.path.exists', return_value=False)
+    def testBuildTemplateApps(self, exists_mock):
```
[the remark this answers is not in the document]
Comment (orangejenny) [reply corehq/apps/app_manager/tests/test_apps.py]: Updated to patch `request` instead in ad9bff5b307

**Comment by orangejenny** [comment]
@gherceg Nit? That's the heart of the test...and it was rotten. Thanks for pointing that out, fixed in f976e61e38b

**Review by gherceg — APPROVED** [review]
Looks good. I might be making this up, but there aren't any issues with asserting within a context manager are there? As long as you've confirmed this test _can_ fail, I feel good about it.

**Comment by orangejenny** [comment]
@gherceg Just tested it out by adding a `self.fail` within the context manager, and it does fail.

**Pull request merged**